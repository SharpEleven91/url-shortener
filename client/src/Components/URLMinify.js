import React, { useState } from "react";
import "../App.css";
import Result from "./Result.js";
import axios from "axios";
import Error from "./Error";
import Logo from "./Logo";
import InputForm from "./InputForm";
import Footer from "./Footer";
const axiosConfig = {
  headers: {
    "Content-Type": "application/json"
  }
};
const baseUrl = "https://umini.herokuapp.com";
export default function URLMinify() {
  const [url, setUrl] = useState('');
  const [result, setResult] = useState('');
  const [error, setError] = useState('');
  function handleChange(event) {
    if (error) {
        setError('');
    }
    setUrl(event.target.value);
  }
  function resetState() {
    setUrl('');
    setResult('');
  }
  function onSubmit() {
    axios
      .post(
        "/api/url-shortener",
        { originalUrl: url, baseUrl: baseUrl },
        axiosConfig
      )
      .then(res => {
        setResult(res.data.shortUrl);
      })
      .catch(err => {
        setError('Bad URL');
      });
  }
  function handleKeyPress(event) {
    if (event.keyCode === 13) {
        onSubmit();
    }
  }
    return (
      <div className="App">
        <Logo />
        <Error checked={error ? true : false} />
        {result ? (
          <Result
            result={result}
            reset={resetState}
          />
        ) : (
          <InputForm
            onSubmit={onSubmit}
            handleKeyPress={handleKeyPress}
            handleChange={handleChange}
          />
        )}
        <>
        <Footer/>
        </>
      </div>
      
    );
}
