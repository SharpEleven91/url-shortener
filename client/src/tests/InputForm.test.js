import { mount, configure } from 'enzyme';
import React from 'react';
import expect from 'expect';
import URLMinify from '../Components/URLMinify';
import { render } from 'react-testing-library';

test('Test', () => {
    const { container } = render(<InputForm/>);
    const text = container.firstChild;
    expect(text.textContent).toBe(" Invalid URL ");
})