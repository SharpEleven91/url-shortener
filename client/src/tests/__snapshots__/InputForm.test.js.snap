// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly 1`] = `
<div
  class="inputgroup"
  data-testid="inputForm"
>
  <div
    class="MuiFormControl-root-6 URLInput-margin-1 URLInput-textField-2"
    data-testid="url-input"
  >
    <div
      class="MuiInputBase-root-23 MuiOutlinedInput-root-10 URLInput-cssOutlinedInput-4 MuiInputBase-formControl-24"
    >
      <fieldset
        aria-hidden="true"
        class="MuiPrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline-17 URLInput-notchedOutline-5"
        style="padding-left: 8px;"
      >
        <legend
          class="MuiPrivateNotchedOutline-legend-41"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input-33 MuiOutlinedInput-input-18"
        id="custom-css-outlined-input"
        placeholder="URL"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root-71 MuiButton-root-45 MuiButton-outlined-53 MuiButton-sizeLarge-69 MuiButton-colorInherit-66 Submit-button-42"
    data-testid="submit-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-46"
    >
       
      MiNiFy
       
    </span>
    <span
      class="MuiTouchRipple-root-74"
    />
  </button>
</div>
`;
